src/Sylius/Bundle/ApiBundle/Serializer/ProductNormalizer.php

```diff
@@ -42,8 +43,12 @@ public function normalize($object, $format = null, array $context = [])
         $context[self::ALREADY_CALLED] = true;
 
         $data = $this->normalizer->normalize($object, $format, $context);
-        $variant = $this->defaultProductVariantResolver->getVariant($object);
-        $data['defaultVariant'] = $variant === null ? null : $this->iriConverter->getIriFromItem($variant);
+
+        $variantsIris = array_map(fn(ProductVariantInterface $variant): string => $this->iriConverter->getIriFromItem($variant), $object->getEnabledVariants()->toArray());
```

Comment:
Can we do this with some custom query? Did you check how to restrict relation data?